src/query/automaton_weight.rs

```diff
@@ -8,28 +8,30 @@ use crate::termdict::{TermDictionary, TermStreamer};
 use crate::DocId;
 use crate::TantivyError;
 use crate::{Result, SkipResult};
+use std::sync::Arc;
 use tantivy_fst::Automaton;
 
 /// A weight struct for Fuzzy Term and Regex Queries
-pub struct AutomatonWeight<A>
-where
-    A: Automaton + Send + Sync + 'static,
```

Comment:
Just curious: did you intentionally remove the `A: Automaton + Send + Sync + 'static` constraint here? What's the implication of this? (I see that the impl block still has a constraint)

Reply:
It has more or less no effects. As far as I can tell, both schools exists in rust.
A lot of people like to never put any boundary on that kind of struct.
I personally like to put it sometimes just to express the intent of the struct.

Here I removed it because it is long... but I could have kept `A: Automaton` for instance.

Reply:
I see. Thanks